vscode/src/localiser.ts

```diff
@@ -29,6 +37,10 @@ class l10Wrapper implements l10n {
       const isPresentInBundle = valueFromBundle !== key;
       return isPresentInBundle ? valueFromBundle : this.defaultTranslation(key, placeholderMap);
     }
+    nbLocaleCode(){
+      const supportedLanguage = vcodeToNbLocaleCodes[vscode.env.language];
+      return supportedLanguage?supportedLanguage:DEFAULT_LANGAUGE;
```

Comment:
It would be better for future support to replace this with a simple function. If required, at a later time, we could use such a mapping only for exceptional cases that do not fit in the standard transformation path.

Further, we do not need to manually set the language to `DEFAULT_LANGUAGE` since the NB backend may support other languages and we don't need to turn off this support. Anyhow, the NB backend will default to english if it does not find a message in the configured language.

So, I'd suggest making `vcodeToNbLocaleCodes = {}` for now, and, change this code block to something like below:
```suggestion
      const vscodeLanguage = vscode.env.language;
      if (!vscodeLanguage) return DEFAULT_LANGUAGE;
      let supportedLanguage:string = vcodeToNbLocaleCodes[vscodeLanguage];
      if (!supportedLanguage) {
          const localeParts = vscodeLanguage.split(/[-_]/g);
          if (localeParts.length > 1) {
              localeParts[1] = localeParts[1].toUpperCase();
          }
          supportedLanguage = localeParts.join(":");
      }
      return supportedLanguage;
```
Thanks.

Reply:
will remove vcodeToNbLocaleCodes for now, if requirement surfaces we can change it. otherwise it becomes funcitonally dead code.